frontend/src/pages/roles/components/ViewRolesWindow.tsx

```diff
@@ -0,0 +1,62 @@
+import RoleDTO from '@shared/dtos/RoleDTO';
+import { useUserContext } from '../../../contexts/UserContext';
+import TabManager from '../../../utility_components/tabs/TabManager';
+import { CreateNewRole } from './CreateNewRole';
+import { ViewPermissions } from './ViewPermissions';
+
+export function ViewRolesWindow({
+    roles,
+    setRoles,
+}: {
+    roles: Record<string, RoleDTO>;
+    setRoles: (roles: Record<string, RoleDTO>) => void;
+}) {
+    const content = Object.keys(roles).map((id) => {
+        return {
+            tabTitle: roles[id].name,
+            tabContent: { ...roles[id] },
+        };
+    });
+
+    const userContext = useUserContext();
+    async function savePermissions(role: RoleDTO) {
+        const data = await userContext.fetcher('PATCH /api/roles/:roleId', role, {
+            roleId: role.id,
+        });
+
+        if (data) {
+            const newRoles = { ...roles, [role.id]: role };
+            setRoles(newRoles);
+        }
+    }
+
+    function isNotCreate(
+        data: typeof content[number] | { tabTitle: 'Create New Role'; tabContent: { isNotTab: true } },
+    ): asserts data is typeof content[number] {
+        if (data.tabTitle === 'Create New Role' && 'isNotTab' in data.tabContent) {
+            throw new Error();
+        }
+    }
+
+    return (
+        <TabManager
+            content={[{ tabTitle: 'Create New Role', tabContent: { isNotTab: true } }, ...content]}
```

Comment:
I really want it to be obvious that "create a new role" is NOT a role, and that it is a button... Not sure what the clean way would be.....

Reply:
Just implemented it take a look. Added an extra loader for tabs for customizability